src/util/heap/layout/vm_layout_constants.rs

```diff
@@ -120,48 +34,130 @@ pub const LOG_PAGES_IN_SPACE64: usize = LOG_SPACE_SIZE_64 - LOG_BYTES_IN_PAGE as
 /** The number of pages in a 64-bit space */
 pub const PAGES_IN_SPACE64: usize = 1 << LOG_PAGES_IN_SPACE64;
 
-/*
- *  The 64-bit VM layout divides address space into LOG_MAX_SPACES (k) fixed size
- *  regions of size 2^n, aligned at 2^n byte boundaries.  A virtual address can be
- *  subdivided into fields as follows
- *
- *    64                              0
- *    00...0SSSSSaaaaaaaaaaa...aaaaaaaa
- *
- * The field 'S' identifies the space to which the address points.
- */
+/// Runtime-initialized virtual memory constants
+#[derive(Clone)]
+pub struct VMLayoutConstants {
```

Comment:
Consider renaming this. This is not a constant any more. `VMLayout` or `VMLayoutParameters` are good candidates.